pkg/virt-controller/watch/preset.go

```diff
@@ -172,6 +179,55 @@ func (c *VirtualMachinePresetController) initializeVirtualMachine(vmi *kubev1.Vi
 	return nil
 }
 
+func applyNamespaceLimitRangeValues(vmi *kubev1.VirtualMachineInstance, limitrangeInformer cache.SharedIndexInformer) {
+	isMemoryFieldExist := func(resource k8sv1.ResourceList) bool {
+		_, ok := resource[k8sv1.ResourceMemory]
+		return ok
+	}
+
+	vmiResources := vmi.Spec.Domain.Resources
+
+	// Copy namespace memory limits (if exists) to the VM spec
+	if vmiResources.Limits == nil || !isMemoryFieldExist(vmiResources.Limits) {
+
+		namespaceMemLimit, err := getNamespaceLimits(vmi.Namespace, limitrangeInformer)
+		if err == nil {
+			if vmiResources.Limits == nil {
+				vmi.Spec.Domain.Resources.Limits = make(k8sv1.ResourceList)
+			}
+			vmi.Spec.Domain.Resources.Limits[k8sv1.ResourceMemory] = *namespaceMemLimit
+		}
+	}
+
+}
+
+func getNamespaceLimits(namespace string, limitrangeInformer cache.SharedIndexInformer) (*resource.Quantity, error) {
+	finalLimit := &resource.Quantity{Format: resource.BinarySI}
+
+	objListKeys := limitrangeInformer.GetStore().ListKeys()
+	// there can be multiple LimitRange values set for the same resource in
+	// a namespace, we need to find the minimal
+	for _, key := range objListKeys {
```

Comment:
Here you should be able to do something like:

```
limits, err := limitrangeInformer.GetIndexer().ByIndex(cache.NamespaceIndex, namespace)
for limit := range limits {
...
}
```

if you make use of the namespace indexer.

Reply:
Nice, thanks! I wasn't aware of this.